src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java

```diff
@@ -189,18 +192,46 @@ private void validateSObjectRangeFilter(String propertyName, Map<ChronoUnit, Int
     }
   }
 
-  private Map<ChronoUnit, Integer> extractRangeValue(String rangeValue) {
+  private Map<ChronoUnit, Integer> extractRangeValue(String propertyName, String rangeValue) {
     if (Strings.isNullOrEmpty(rangeValue)) {
       return Collections.emptyMap();
     }
     return Stream.of(rangeValue.split(","))
       .map(String::trim)
       .map(s -> s.split(":", 2))
-      .filter(keyValue -> keyValue.length > 1)
+      .peek(keyValue -> validateUnitKeyValue(propertyName, rangeValue, keyValue))
       .collect(Collectors.toMap(
-        keyValue -> ChronoUnit.valueOf(keyValue[1].trim().toUpperCase()),
-        keyValue -> Integer.parseInt(keyValue[0].trim()),
+        keyValue -> parseUnitType(propertyName, keyValue[1]),
+        keyValue -> parseUnitValue(propertyName, keyValue[0]),
         (o, n) -> n
       ));
   }
+
+  private void validateUnitKeyValue(String propertyName, String rangeValue, String[] keyValue) {
+    if (keyValue.length < 2) {
+      throw new InvalidConfigPropertyException(
+        String.format("Range property '%s' value has invalid format '%s'. "
+                        + "Expected format is <VALUE_1>:<TYPE_1>,<VALUE_2>:<TYPE_2>... . "
+                        + "For example, '1:DAYS,2:HOURS,30:MINUTES'", propertyName, rangeValue), propertyName);
```

Comment:
This format doesn't match what is in the docs (1 days, 2 hours, 30 minutes). I think the space is more friendly than colon and should be used instead